website/docs/language/functions/index.mdx

```diff
@@ -33,3 +32,36 @@ the OpenTofu expression console, by running
 
 The examples in the documentation for each function use console output to
 illustrate the result of calling the function with different parameters.
+
+## Provider Defined Functions
+
+As of OpenTofu 1.7.0, providers may define their own functions to be available during
+execution.
+
+OpenTofu iterates through the [required_providers](/docs/language/providers/requirements/) block
+and queries the specified providers for any functions they wish to register. Functions are
+added to the current module's context under `provider::<provider_name>::<function_name>`.
+Functions are scoped to the module that requires the provider and are not inherited by child modules.
+
+### Example:
+
+```hcl
+terraform {
+  required_providers {
+    myhelper = {
+      # yantrio/helpers registers a function named "echo"
+      source = "yantrio/helpers"
+    }
+  }
+}
+
+locals {
+  myval = provider::myhelper::echo("Hello Functions!")
+}
+```
+
+
+### Notes for Provider Authors:
+* Support for functions was added in protocol version 5.5 and 6.5
+* OpenTofu's provider protocol is compatible with Hashicorp's provider protocol
```

Comment:
Hashicorp's => Terraform's?

Cause I believe other HashiCorp projects also have providers, and we explicitly mean Terraform here.